v2/pkg/protocols/common/executer/executer.go

```diff
@@ -107,7 +107,10 @@ func (e *Executer) Execute(input *contextargs.Context) (bool, error) {
 			if !event.HasOperatorResult() && !event.UsesInteractsh {
 				lastMatcherEvent = event
 			} else {
-				if writer.WriteResult(event, e.options.Output, e.options.Progress, e.options.IssuesClient) {
+				if !(event.UsesInteractsh && event.InteractshMatched.Load()) && writer.WriteResult(event, e.options.Output, e.options.Progress, e.options.IssuesClient) {
+					if event.UsesInteractsh {
+						event.InteractshMatched.CompareAndSwap(false, true)
```

Comment:
Since we are not interested in the previous value or anyhow the result of comparison, I think this is equivalent to `.Set(true)` which should skip some comparison overhead. Probably the same reasoning can be applied in other places where the return value of the function is not used. What do you think?

Reply:
done